### PR TITLE
fix: Continuation line cause a semantic error

### PR DESCRIPTION
**Pull request by ap891843**
Signed-off-by: ap891843 <aman.prashant@broadcom.com>

### CODE REVIEW

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java [hunk 1]
```diff
@@ -399,14 +403,35 @@ private Locality retrievePosition(@NonNull ParserRuleContext ctx) {
...
+      Position startPosition;
+      Position stopPosition;
+      if (StringUtils.isNotBlank(token.getText())
+          && positionCorrectionMap.containsKey(token.getLine() - 1)
+          && positionCorrectionMap
+              .get(token.getLine() - 1)
+              .getContentArea()
+              .contains(token.getText())) {
+        CobolLine continueCobolLine = positionCorrectionMap.get(token.getLine() - 1);
+        int horizontalCoordinate = continueCobolLine.getContentArea().indexOf(token.getText()) + 7;
```
Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:416]: just optional: meaningful const, COMMENT_POSITION = 7

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java [hunk 2]
```diff
@@ -399,14 +403,35 @@ private Locality retrievePosition(@NonNull ParserRuleContext ctx) {
...
-            .uri(documentUri)
-            .copybookId(retrieveCopybookId())
-            .range(new Range(getStartPosition(token), getStopPosition(token)))
-            .token(token.getText())
-            .recognizer(GrammarPreprocessorListenerImpl.class)
-            .build();
+    return token -> {
+      Position startPosition;
+      Position stopPosition;
+      if (StringUtils.isNotBlank(token.getText())
```
Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:409]: optional: can be extracted to a validation method, regarding to positionCorrectionMap

**Review by asatklichov — APPROVED** [review]
:electron:

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java (hunk 1 above)
Comment (ap891843) [reply server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java]: Done.

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java (hunk 2 above)
Comment (ap891843) [reply server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java]: done.

**Comment by temanbrcom** [comment]
Please, add the issue number to the commit message

**Comment by ap891843** [comment]
> Please, add the issue number to the commit message

Done

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java [hunk 3]
```diff
@@ -63,6 +64,7 @@
 public class GrammarPreprocessorListenerImpl extends CobolPreprocessorBaseListener
     implements GrammarPreprocessorListener {
 
+  private static final int AREA_A_POSITION = 7;
```
Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:67]: Area A position is 8-11, so please rename this to "INDICATOR_POSITION"

Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:67]: Area A position is 8-11, so please rename this to "INDICATOR_POSITION"

Comment (ap891843) [reply server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:67]: Nice! 👍

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessor.java [hunk 4]
```diff
@@ -41,4 +43,24 @@
...
+   * @param uri - the URI of the processing document
+   * @param code - COBOL program text to analyse
+   * @param copybookStack - stack, representing the hierarchy of nesting copybooks
+   * @param copybookProcessingMode - reflects the status of the copybook support (ENABLED|DISABLED)
+   * @param positionCorrectionMap - contains mapping between the modified new line in document as
+   *     per preprocessing to the actual {@link CobolLine} in the document
+   * @return extended document with copybooks and related errors
+   */
+  @NonNull
+  ResultWithErrors<ExtendedDocument> buildExtendedDocument(
```
Comment (temanbrcom) [inline server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessor.java:60]: Are you using both of those methods?

Comment (ap891843) [reply server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessor.java:60]: yes, we may get rid of one which calls the other one.

File: server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java [hunk 5]
```diff
@@ -399,14 +404,40 @@ private Locality retrievePosition(@NonNull ParserRuleContext ctx) {
...
-            .copybookId(retrieveCopybookId())
-            .range(new Range(getStartPosition(token), getStopPosition(token)))
-            .token(token.getText())
-            .recognizer(GrammarPreprocessorListenerImpl.class)
-            .build();
+    return token -> {
+      Position startPosition;
+      Position stopPosition;
+      if (isTokenRepositioned(token)) {
+        CobolLine continueCobolLine = positionCorrectionMap.get(token.getLine() - 1);
```
Comment (temanbrcom) [inline server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:411]: What will happen if the token follows a literal that consists of more than one continuation line?

Comment (ap891843) [reply server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java:411]: Correct. Logic fails here. I will draft this PR.

**Pull request closed without merging**